### ISSUE TITLE
Speech-based code editor

### ISSUE THREAD

**Issue by amyjko** [wordplaydev/wordplay#1270]
### What are you trying to do that you can't?

For people who cannot do drag and drop or type on a keyboard, and find single-character on-screen keyboards slow, the current code editor is a major barrier to writing code.

### What is your idea?

Let's imagine a speech-based code editor. Browsers support speech to text, so the task is designing an editor that takes the transcribed text and uses it as an interface to all existing editing commands.

One possibility is that we define templated multilingual text patterns for all commands, and an interface for someone dictating to see all commands, so they can learn the patterns of communication. We could also have system for taking an expression and finding the best guesses of which command was intended, then a quick way to select the synthesized match. The expression language could use relative spatial references (e.g., 'here').
